### PR TITLE
allow choosing between gpt-3.5-finetuned and gpt-4

### PR DESCRIPTION
**Pull request by rmuller-ml**

### CODE REVIEW

File: server/bleep/src/agent.rs [hunk 1]
```diff
@@ -36,6 +36,23 @@ mod tools {
...
+    model_name: &'static str,
+
+    /// The number of tokens reserved for the answer
+    headroom: usize,
+}
+
+pub const GPT3_FINETUNED: AnswerModel = AnswerModel {
+    tokenizer: "gpt-3.5-turbo-0613",
+    model_name: "gpt-3.5-finetuned", // ft:gpt-3.5-turbo-0613:bloop::7tkKlNOw
+    headroom: 512,
```
Comment (rmuller-ml) [inline server/bleep/src/agent.rs:53]: We need prompt and answer headroom, for GPT4 and GPT3.

And make it configurable so the front end can change it.

File: server/bleep/src/agent/model.rs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    /// The number of tokens reserved for history
+    pub history_headroom: usize,
+}
+
+pub const GPT_3_5_FINETUNED: AnswerModel = AnswerModel {
+    tokenizer: "gpt-3.5-turbo-0613",
+    model_name: "gpt-3.5-finetuned",
+    answer_headroom: 512,
+    prompt_headroom: 1250,
```
Comment (rmuller-ml) [inline server/bleep/src/agent/model.rs:25]: prompt headroom should be >1600

File: server/bleep/src/agent/model.rs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+pub const GPT_4: AnswerModel = AnswerModel {
+    tokenizer: "gpt-4-0613",
+    model_name: "gpt-4-0613",
+    answer_headroom: 1024,
+    prompt_headroom: 2500,
+    history_headroom: 2048,
+};
+
+#[allow(clippy::wildcard_in_or_patterns)]
```
Comment (rsdy) [inline server/bleep/src/agent/model.rs:37]: can you move this lint onto the `match` or the `fn`?

**Review by rsdy — APPROVED** [review]

**Review by rmuller-ml — COMMENTED** [review]
LGTM

**Review by ggordonhall — APPROVED** [review]

**Pull request merged**